### PR TITLE
Add QuickNet v1 architecture

### PR DESCRIPTION
**Pull request by lgeiger**
This PR adds a new version of QuickNet. I also renamed to models to `QuickNetSmall`, `QuickNet` and `QuickNetLarge` and added the simple training loop used to train the models.

### CODE REVIEW

**Comment by lgeiger** [comment]
This is now ready for review. I also manually verified that reloaded QuickNet reaches the correct ImageNet accuracy.

File: larq_zoo/sota/quicknet.py [hunk 1]
```diff
@@ -1,69 +1,56 @@
...
-
-
-class QuickNetBaseFactory(ModelFactory, abc.ABC):
-    name: str = "model"
-    blocks_per_section: Sequence[int] = Field()
-    section_filters: Sequence[int] = Field()
-    use_squeeze_and_excite_in_section: Sequence[bool] = Field()
-    transition_block: Callable[..., tf.Tensor] = Field()
-    stem_filters: int = Field(64)
+def blurpool_initializer(shape, dtype=None):
```
Comment (leonoverweel) [inline larq_zoo/sota/quicknet.py:12]: Can you add a docstring + reference here?

**Review by leonoverweel — APPROVED** [review]
Thanks for adding this, just one small request.

**Comment by leonoverweel** [comment]
Should we do a Zoo release once this is merged and propagate these new numbers to the docs as well?

**Comment by lgeiger** [comment]
> Should we do a Zoo release once this is merged and propagate these new numbers to the docs as well?

Yes, I think we should do a major release and update the docs

**Pull request merged**